share/wake/lib/system/path.wake

```diff
@@ -96,33 +96,17 @@ export def basename (file: String): String =
 export def dirname (file: String): String =
     simplify "{file}/.."
 
-export def isPathDir: Path => Boolean = match _
-  BadPath _ = False
-  Path str = matches `0+` (hashcode str)
+export def isPathDir (path: Path): Boolean =
+    getPathHash path
+    | matches `0+`
 
 # Paths differ from Strings in that they have been hashed; their content is frozen
-from internals_wake export type Path
-from internals_wake import def Path BadPath # Path constructor stays private!
+export data Path =
+    Path (name: String)
 
-export def getPathResult (path: Path): Result Path Error = match path
-  BadPath e = Fail e
-  _         = Pass path
-
-export def getPathError (path: Path): Option Error = match path
-  Path _    = None
-  BadPath e = Some e
-
-export def getPathName (path: Path): String = match path
-  Path name = name
-  BadPath _ = "BADPATH"
-
-export def getPathChild (path: Path): String = match path
-  Path name = replace `^.*/` '' name
-  BadPath _ = "BADPATH"
-
-export def getPathParent (path: Path): Path = match path
```

Comment:
This cannot be replicated using `mkdir`, because the parent might have been created by a `Job`, resulting in two jobs with the same output file. If you run `getPathParent` on `Path "."`, you will get `Path ".."`, which I don't think is too terrible.

So, please don't remove this one.

Reply:
Isn't `mkdir` an exception to the same-output checks?  Anyway, if you don't see an issue with this (and since even `/..` is a valid absolute path) I'll add it back in.

Reply:
Multiple to `mkdir` do not conflict with each other, but would conflict with a `Job`.
Actually, `simplify "/.." = "/"`.